### DISCUSSION TITLE
Unable to resolve LSP4E target following a clean installation

### DISCUSSION THREAD

**Discussion by joaodinissf** [eclipse/lsp4e#849] [Q&A] [answered]
Hello,

I'm trying to set up an Eclipse workspace that would allow me to contribute to LSP4E.

I have followed [these instructions](https://github.com/eclipse/lsp4e/blob/master/CONTRIBUTING.md#%EF%B8%8F-setting-up-the-development-environment) to perform a clean installation of "Eclipse IDE for Eclipse Committers". In step 3, I chose to set the provided target file, `target-platforms/target-platform-latest/target-platform-latest.target`, as the active target platform.

Unfortunately, at this point, Eclipse isn't able to resolve the required dependencies. Inspecting the target definition file reveals the following issues:

![image](https://github.com/eclipse/lsp4e/assets/6786818/32729fca-88d2-4ef8-8e72-6980bc48e96d)

I tried to use the information from these error messages to manually patch the file, to no avail. Nevertheless, it seems that something has gone wrong during the setup process. Does anyone have an idea of what that might have been?

Thanks in advance.

**Comment by rubenporras** [comment, marked as answer]
Hi,

I can reproduce the problem in a new installation as well, but I do not know how to fix it.

It looks like a bug in PDE, as the target file works fine in our builds that use Tycho.

I also found a very similar error in https://marketplace.eclipse.org/content/error/report/5660877.

@mickaelistria , @laeubi, any ideas?

Thanks

**Reply by laeubi** [reply]
@rubenporras this is not a bug but more a missing features:

- https://github.com/eclipse-pde/eclipse.pde/issues/207

I'll try to work on this in the next days, but don't expect ti to aproach before 2023-12 release ....

**Comment by rubenporras** [comment]
Hi @joaodinissf,

I can make it work by removing the maven section of the target file replacing it by a normal p2 orbit repository:

```
  <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="true" type="InstallableUnit">
    <repository location="https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/2023-09"/>
    <unit id="com.google.guava" version="32.1.2.jre"/>
    <unit id="com.google.guava.failureaccess" version="1.0.1"/>
  </location>

```

@mickaelistria , I suggest, that until the workbench is fixed we commit the change I propose in the repository. What do you think?

Regards

**Reply by mickaelistria** [reply]
Whatever works ;)

**Reply by joaodinissf** [reply]
Just wanted to confirm that @rubenporras's suggestion is a valid workaround until the underlying issue gets fixed. Thanks!
